### ISSUE TITLE
QuotesBear broken by Python triple quotes

### ISSUE THREAD

**Issue by underyx** [coala/coala-bears#1103]
This is a bad suggestion:

```diff
 class StructlogBear(LocalBear):
-    """Checks for correct usage of structlog"""
+    '""Checks for correct usage of structlog""'
```

type/bug

**Comment by sils** [comment]
oh yes, @abhsag24 was working on fixing this already

**Comment by netman92** [comment]
What's the status here ? @rohanraju97

**Comment by sils** [comment]
I think @abhsag24 has ongoing work on this?

**Comment by rohanraju97** [comment]
I worked on this for a while, but then started working on my gsoc project. I will do it once the application date for gsoc is crossed. If someone wants to take care of this issue in the meantime, I'll unassign myself. @sils I didn't know that. The issue didn't have any assignees when I saw it so I assigned it to myself.

**Comment by abhsag24** [comment]
I did. Then @aptrishu took it into his own PR idk if it solves it
completely but it's a step in this direction


On 16-Mar-2017 1:17 PM, "Rohan Raju" <notifications@github.com> wrote:

I worked on this for a while, but then started working on my gsoc project.
I will do it once the application date for gsoc is crossed. If someone
wants to take care of this issue in the meantime, I'll unassign myself.
@sils <https://github.com/sils> I didn't know that. The issue didn't have
any assignees when I saw it so I assigned it to myself.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/coala/coala-bears/issues/1103#issuecomment-286981483>,
or mute the thread

**Comment by aptrishu** [comment]
Ok, I'll look into it, if my PR already solves it or not.

**Comment by aptrishu** [comment]
https://github.com/coala/coala-bears/issues/1024 relevant ? @abhsag24

**Comment by abhsag24** [comment]
yup this happens since annotationBear doesnt diferentiate between single line and multiline strings, It should get solved when  it correctly identifies.

**Comment by sils** [comment]
@abhsag24 state of this? This needs to get fixed :/

**Comment by aptrishu** [comment]
with the new annotationBear, it can identify singleline and multiline strings clearly. So, that should solve this problem.

**Comment by meetmangukiya** [comment]
bump bump bump, I hit this again... it seems broken only on single line triple quotes

**Comment by sils** [comment]
you can speed this up by reviewing
https://github.com/coala/coala-bears/pull/1332

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala.io/ - http://viperdev.io/ - http://gitmate.com/

**Comment by netman92** [comment]
What's the status here, please? @rohanraju97 @sils

**Comment by aptrishu** [comment]
cc @userzimmermann , new AnnotationBear will fix it.

**Comment by userzimmermann** [comment]
@aptrishu We can go with https://github.com/coala/coala-bears/pull/1332 first and later think again about `pygments` and https://github.com/coala/coala-bears/pull/2096 for some simplification... As long as the `AnnotationBear` does what it should :)

cc @netman92
